homeassistant/helpers/template.py

```diff
@@ -1424,6 +1425,33 @@ def state_attr(hass: HomeAssistant, entity_id: str, name: str) -> Any:
     return None
 
 
+def has_value(hass: HomeAssistant, entity_id: str | State | Iterable | None) -> bool:
+    """Test if an entity has a valid value."""
+    # Entity is a non-existing object
+    if entity_id is None:
+        return False
+
+    # If entity is a string or object then make it a list first.
+    if isinstance(entity_id, (str, State)):
+        entity_id = [entity_id]
```

Comment:
This is not a  pattern we generally have in our template methods, and I'm not sure if that is a good idea (as this would also mean we should implement it consistently everywhere else?)

Let's start with keeping things simple and just accept a single entity ID.

Reply:
@frenck , I did that to allow something like this:
```
{{{ ['<entity1>', '<entity2>', '<entity3>'] | select("has_value") | first | default }}
```
This to make it easier if one has a few integrations retrieving the same item and then to get the one that is available.
For example, having a local integration but also for the same thing a cloud integration. This way one can prefer the local one but if for some reason it is failing try the cloud one.
For me, I have an integration that polls for my car and another integration for a cloud service that also has the data on my car. I prefer the one that polls my car, but if it fails for some reason then it falls back to the one from the cloud service.
But I can remove if you prefer.

Reply:
I know your intention, however, as said, this is not normal handling compared to other methods.